packages/firebase_auth/test/firebase_auth_test.dart

```diff
@@ -220,7 +232,8 @@ void main() {
         link: '<Url with domain from your Firebase project>',
       );
       FirebaseUser user = await auth.currentUser();
-      user = await user.linkWithCredential(credential);
+      AuthResult result = await user.linkWithCredential(credential);
+      user = result.user;
```

Comment:
In some cases the `user` is extracted in one statement and in other cases it is extracted in two statements. Maybe it would be more readable to keep it consistent.

Reply:
Good call. I've updated the tests. In the process I wrote some more tests of the new code I added and fixed some issues (I had missed reauthenticate on my first pass, and the getter for profile lacked an appropriate cast)